### PR TITLE
Firmware info on setup tab

### PR DESCRIPTION
**Pull request by HThuren**
After, with possibility to future additions of info from MCU in system info.
Now MSP API version, Build date and Build options for firmware build.
image

Before
image

### CODE REVIEW

File: locales/en/messages.json
Comment (haslinghuis) [inline locales/en/messages.json:1008]: API :)

**Comment by haslinghuis** [comment]
We also could expose build key and build options

**Comment by HThuren** [comment]
> We also could expose build key and build options

Yes, they seems to be empty (or are this becourse I use own build)

File: locales/en/messages.json
Comment (HThuren) [reply locales/en/messages.json]: Changed

**Comment by HThuren** [comment]
@haslinghuis 
I installed a cloud build firmware, but nothing. Maybe I look at a wrong place, where (variables in msp) do you think I have to look ?

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by haslinghuis** [comment]
MSP2_GET_TEXT

See also `https://github.com/betaflight/betaflight-configurator/pull/3332`

**Comment by HThuren** [comment]
@haslinghuis you got your BuildOptions now :-)

**Comment by haslinghuis** [comment]
@blckmn check this :point_up:

File: locales/en/messages.json
Comment (haslinghuis) [inline locales/en/messages.json:1030]: Maybe Custom Build - or we just hide the section in that case.
What does Core Build look like?

Comment (HThuren) [reply locales/en/messages.json]: coreBuild are "Core Only"
I think for initialSetupInfoBuildOptionsEmpty "Local Build - no Defines" are right - changed to that

File: src/js/tabs/setup.js
Comment (haslinghuis) [inline src/js/tabs/setup.js:281]: Removing one sonar cloud issues (we can ignore the bugs in these case)
```suggestion
                    for (const buildOption of FC.CONFIG.buildOptions) {
                        build_opt_e.append(buildOption, ' ');
```

Comment (HThuren) [reply src/js/tabs/setup.js]: great change, done

**Review by haslinghuis — APPROVED** [review]

**Comment by nerdCopter** [comment]
i'd personally prefer to see system and firmware informoation on top, but others may have differing opinion.

**Comment by HThuren** [comment]
> * i'd personally prefer to see system and firmware information on top, but others may have differing opinion.
> * tested successfully

In fact I tried to move the instrument part to right top , but didn't success...
The idea are maybe more information can come, then the information part will grow.

**Comment by HThuren** [comment]
Right, now with buildkey and sensors resolved to hardware names.
But I need help to test sensors names, since my hw don't respone with hardware numbers, or maybe I have a mistake in the code

**Comment by nerdCopter** [comment]
yesterday i was typing a suggestion for build-key as well, but ended up not submitting. now i see you coded it :heart_decoration: .  Might i suggest it link-able to `https://build.betaflight.com/api/builds/[build-key]/log` and/or `https://build.betaflight.com/api/builds/[build-key]/json`?

**Comment by HThuren** [comment]
@nerdCopter - you got your config and log :-)

**Review by nerdCopter — APPROVED** [review]

**Comment by HThuren** [comment]
Great, thank you

**Comment by nerdCopter** [comment]
@HThuren , needs css fix or proper class assignment to buttons.

nearly invisible in dark-mode.
image

and seems to be blue (link color) text on light mode.
image


brought to my attention my @ Zzyzx in discord

**Comment by HThuren** [comment]
> @HThuren , needs css fix or proper class assignment to buttons.
> 
> nearly invisible in dark-mode. image
> 
> and seems to be blue (link color) text on light mode. image
> 
> brought to my attention my @ Zzyzx in discord

Argh, I didn't think of the dark mode, I go to fix.
I also pack the Build options into , maybe just the title to a button, the precentation right now are not nice

**Pull request merged**
